### PR TITLE
Add E0763

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/test/ui/parser/byte-literals.rs [hunk 1]
```diff
@@ -8,5 +8,5 @@ pub fn main() {
     b'	';  //~ ERROR byte constant must be escaped
     b''';  //~ ERROR byte constant must be escaped
     b'é';  //~ ERROR byte constant must be ASCII
-    b'a  //~ ERROR unterminated byte constant
+    b'a  //~ ERROR E0763
```
Comment (petrochenkov) [inline src/test/ui/parser/byte-literals.rs:11]:
```suggestion
    b'a  //~ ERROR [E0763]: unterminated character literal
```
Could you avoid removing the human readable annotation?

File: src/librustc_parse/lexer/mod.rs [hunk 2]
```diff
@@ -339,8 +339,15 @@ impl<'a> StringReader<'a> {
             }
             rustc_lexer::LiteralKind::Byte { terminated } => {
                 if !terminated {
-                    self.fatal_span_(start + BytePos(1), suffix_start, "unterminated byte constant")
-                        .raise()
+                    self.sess
+                        .span_diagnostic
+                        .struct_span_fatal_with_code(
+                            self.mk_sp(start + BytePos(1), suffix_start),
+                            "unterminated character literal",
```
Comment (petrochenkov) [inline src/librustc_parse/lexer/mod.rs:346]:
```suggestion
                            "unterminated byte literal",
```

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**